atlasdb-client/src/main/java/com/palantir/atlasdb/stream/AbstractGenericStreamStore.java

```diff
@@ -210,9 +211,11 @@ private void loadNBlocksToOutputStream(
             long firstBlock,
             long numBlocks,
             OutputStream os) {
-        for (long i = 0; i < numBlocks; i++) {
-            loadSingleBlockToOutputStream(tx, streamId, firstBlock + i, os);
+        long position = 0;
+        for (; position + BLOCK_BATCH <= numBlocks; position += BLOCK_BATCH) {
```

Comment:
Are you sure this is right? The batching should be controlled outside of this method. It is actually a parameter to the codegen, so you should not invent a new constant here.

Reply:
I know it's parameterised in the code-gen too, but I don't think you want to pass for example 100 blocks to that method, as it will get you all 100 blocks at once. This is the closest thing to what the old gen did, but for 2 blocks at once (although it's entirely possible that I'm misunderstanding something along the way).

Reply:
My point is that this method should be doing exactly what it's being asked to do (so if it's 100 blocks, it's 100 blocks) and the users control the batching somewhere else. There's efficient batching below this method too, so it should not be making batching decisions.

Reply:
the first pass, and then even with this at 4, caused SLOs to fail (although not at this level) while upgrading atlasdb-proxy.

The "user" that consumes this is `BlockGetter` if I'm not mistaken - so maybe the point is that the methods there need to change it, but by the time the user gets the result from this, I'm pretty certain that it's just an `InputStream` that they don't know or care about how many blocks are being loaded - that's the responsibility of code in this area (whether this class or `BlockGetter` or one of the other classes on that path), I'm not sure, but it should surely amount to roughly the same.